packages/ui/src/components/Form/bean/NewBeanModal.tsx

```diff
@@ -17,13 +18,19 @@ export type NewBeanModalProps = {
 export const NewBeanModal: FunctionComponent<NewBeanModalProps> = (props: NewBeanModalProps) => {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   const [beanModel, setBeanModel] = useState<any>();
+  const submitRef = useRef<CustomAutoFormRef>(null);
 
   useEffect(() => {
     setBeanModel(props.beanName ? { name: props.beanName } : {});
   }, [props.beanName]);
 
-  const handleConfirm = useCallback(() => {
-    props.onCreateBean(beanModel as RegistryBeanDefinition);
+  const handleConfirm = useCallback(async () => {
+    // validation updates the bean model, so we need to clone it to avoid creating the bean with default values
+    const beanModelTmp = { ...beanModel };
+    const valid = await submitRef.current?.form.validate();
+    if (valid === undefined || valid === null) {
```

Comment:
[pro-tip]: We have the [`isDefined`](https://github.com/KaotoIO/kaoto-next/blob/a6f24641dac3cc646b9da767ad6c66c81b766dbd/packages/ui/src/utils/is-defined.ts) util to validate these two options at once